cmd/mavenBuild.go

```
@@ -133,84 +174,96 @@ func runMavenBuild(config *mavenBuildOptions, telemetryData *telemetry.CustomDat
+				if err != nil {
+					return err
 				}
 
-				var buildArtifacts build.BuildArtifacts
-
-				buildArtifacts.Coordinates = buildCoordinates
-				jsonResult, _ := json.Marshal(buildArtifacts)
-				commonPipelineEnvironment.custom.mavenBuildArtifacts = string(jsonResult)
+				if config.CreateBuildArtifactsMetadata {
+					err2, done := createBuildArtifactsMetadata(config, commonPipelineEnvironment)
+					if done {
+						return err2
+					}
+				}
+				return nil
 			}
-
-			return nil
 		} else {
 			log.Entry().Infof("publish not detected, ignoring maven deploy")
 		}
 	}
-
 	return err
 }
 
-func getPurlForThePomAndDeleteIndividualBom(pomFilePath string) string {
-	bomPath := filepath.Join(filepath.Dir(pomFilePath) + "/target/" + mvnBomFilename + ".xml")
+func createBuildArtifactsMetadata(config *mavenBuildOptions, commonPipelineEnvironment *mavenBuildCommonPipelineEnvironment) (error, bool) {
+	fileUtils := &piperutils.Files{}
+	buildCoordinates := []versioning.Coordinates{}
+	options := versioning.Options{}
+	var utils versioning.Utils
+
+	matches, _ := fileUtils.Glob("**/pom.xml")
+	for _, match := range matches {
+
+		artifact, err := versioning.GetArtifact("maven", match, &options, utils)
+		if err != nil {
+			log.Entry().Warnf("unable to get artifact metdata : %v", err)
+		} else {
+			coordinate, err := artifact.GetCoordinates()
+			if err != nil {
+				log.Entry().Warnf("unable to get artifact coordinates : %v", err)
+			} else {
+				coordinate.BuildPath = filepath.Dir(match)
+				coordinate.URL = config.AltDeploymentRepositoryURL
+				coordinate.PURL = getPurlForThePom(match)
+				buildCoordinates = append(buildCoordinates, coordinate)
+			}
+		}
+	}
+
+	if len(buildCoordinates) == 0 {
+		log.Entry().Warnf("unable to identify artifact coordinates for the maven packages published")
+		return nil, true
+	}
+
+	var buildArtifacts build.BuildArtifacts
+
+	buildArtifacts.Coordinates = buildCoordinates
+	jsonResult, _ := json.Marshal(buildArtifacts)
```

ignoring errors is not recommended